### PR TITLE
all done with UCSBDiningCommonsMenuItem

### LINKED ISSUE #27: Add `DELETE` endpoint for a specific record in UCSBDiningCommonsMenuItem table.
Add `DELETE` endpoint for a specific record in UCSBDiningCommonsMenuItem table.

# Acceptance Criteria:

- [ ] In `UCSBDiningCommonsMenuItemController.java` there is code for an 
      endpoint `DELETE /api/UCSBDiningCommonsMenuItem?id=123` endpoint 
      that deletes the record if it exists, and returns 200 (ok) and 
      the text `record 123 deleted`, or returns 404 (Not Found) and
      the text `record 123 not found` if it does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBDiningCommonsMenuItemController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBDiningCommonsMenuItemController.java`

### LINKED ISSUE #28: Add `PUT` (edit) endpoint for a single record in UCSBDiningCommonsMenuItem table
Add `PUT` (edit) endpoint for a single record in UCSBDiningCommonsMenuItem table

Note that it is a good idea to implement `GET` for a single item
first; when using `PUT` manually, it's convenient to be able to copy/paste the JSON representation of the data and then edit it in the swagger interface.

Eventually, it's typically the case that in order to populate an edit
form in the front end, you first do a `GET` on a particular item, 
and use that JSON to populate the fields in a form.  You can send the
JSON representation of the new item to the backend.

# Acceptance Criteria:

- [ ] In `UCSBDiningCommonsMenuItemController.java` there is code for an 
      endpoint `PUT /api/UCSBDiningCommonsMenuItem?id=123` endpoint 
      that accepts JSON for a new set of values for the database
      fields other than `id`, and updates the values of those fields.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBDiningCommonsMenuItemController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBDiningCommonsMenuItemController.java`

### LINKED ISSUE #29: Add `GET` (show) endpoint for a single record in UCSBDiningCommonsMenuItem table
Add `GET` (show) endpoint for a single record in UCSBDiningCommonsMenuItem table

# Acceptance Criteria:

- [ ] In `UCSBDiningCommonsMenuItemController.java` there is code for an 
      endpoint `GET /api/UCSBDiningCommonsMenuItem?id=123` endpoint 
      that returns the JSON of the database record with id 123 if it
      exists, or a 400 and the error message `id 123 not found` if it
      does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBDiningCommonsMenuItemController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBDiningCommonsMenuItemController.java`

### LINKED ISSUE #30: Create `UCSBDiningCommonsMenuItemController`, add `GET` (index) and `POST` (create)
Create `UCSBDiningCommonsMenuItemController`, add `GET` (index) and `POST` (create)

# Acceptance Criteria:

- [ ] There is a controller file `UCSBDiningCommonsMenuItemController.java`
      in the expected directory.
- [ ] In `UCSBDiningCommonsMenuItemController.java` there is 
      code for a `GET /api/UCSBDiningCommonsMenuItem/all` endpoint 
      that returns a JSON list of all subreddits in the database.
      (We sometimes call this an *index* action since it lists all
      items in the database.)
- [ ] In `UCSBDiningCommonsMenuItemController.java` there is 
      code for a `POST /api/UCSBDiningCommonsMenuItem/post` endpoint
      that can be used to create a new entry in the table. (This
      is a *create* action.)
- [ ] The Swagger-UI endpoints for these are well documented so that
      any member of the team can understand what they are for and
      how to use them.
- [ ] The `POST` endpoint works as expected, in the sense that new
      records can be added to the database (on localhost).
- [ ] The `GET` endpoint works as expected, in the sense that the new
      records that are added show up (on localhost).
- [ ] The `GET` and `POST` endpoints work as expected when the 
      app is deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the methods in 
      `UCSBDiningCommonsMenuItemController.java`
- [ ] There is full mutation test coverage (Pitest) for the methods in
      `UCSBDiningCommonsMenuItemController.java`

### LINKED ISSUE #31: Create database table for UCSBDiningCommonsMenuItem
Create database table for UCSBDiningCommonsMenuItem

# Acceptance Criteria:

- [ ] There is an `@Entity` class called UCSBDiningCommonsMenuItem.java
- [ ] There is a `@Repository` class called UCSBDiningCommonsMenuItemRepository.java
- [ ] When you start up the repo on localhost, you can see the table
      using the H2 console (see the file `docs/h2-database.md` for 
      instructions.)
- [ ] When you deploy the app to Heroku, you can see the table
      using the psql command (See `Accessing Database Console` in the
      main `README.md` file for instructions.)

### PR DESCRIPTION
**Pull request by haoyuanli2023**
I finished eveything for the Dining Commons Menu Item, and my code passed the tests. Hope you can see if it is good enough to merge in our main! Thanks team! Closes #27 Closes #28 Closes #29 Closes #30 Closes #31

### CODE REVIEW

**Comment by DistractedPrimate** [comment]
This looks good! Can you link the issues that this pull request closes? Good work

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBDiningCommonsMenuItemController.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("")
+    public UCSBDiningCommonsMenuItem updateItem(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid UCSBDiningCommonsMenuItem incoming) {
+
+        UCSBDiningCommonsMenuItem item = ucsbDiningCommonsMenuItemRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(UCSBDiningCommonsMenuItem.class, id));
+
+        item.setDiningCommonsCode(incoming.getDiningCommonsCode());
```
Comment (roberto-garcia-0) [inline src/main/java/edu/ucsb/cs156/example/controllers/UCSBDiningCommonsMenuItemController.java:95]: Mutation coverage does not cover line 95 and 96 in the controller for the `PUT` endpoint.

**Review by roberto-garcia-0 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBDiningCommonsMenuItemController.java (hunk 1 above)
Comment (roberto-garcia-0) [reply src/main/java/edu/ucsb/cs156/example/controllers/UCSBDiningCommonsMenuItemController.java:95]: @haoyuanli2023 Try to put in authorization tests for the `PUT` endpoint and the `DELETE` endpoint.

Comment (DistractedPrimate) [reply src/main/java/edu/ucsb/cs156/example/controllers/UCSBDiningCommonsMenuItemController.java:95]: fixed put test so now It works

**Pull request merged**